source/blog/039-rfc-package-importer.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Request for Comments: Package Importer"
+author: James Stuckey Weber
+date: 2023-09-30 14:30:00 -8
+---
+
+Sass users often need to use styles from a dependency to customize an existing
+theme or access styling utilities. Historically, Sass has not specified a
+standard method for using packages from dependencies. This has led to a variety
+of domain-specific solutions, including the `~` prefix in Webpack, and adding
+`node_modules` to `loadPaths`.
+
+This has been a common pain point, and can make it difficult to rely on
+dependencies. It can also make it more difficult to move your project to a new
+build process.
+
+## Package Importers
+
+We are proposing a new type of importer that allows users to use the `pkg:` URL
+scheme to direct Sass to resolve the dependency URL using the resolution
+standards and conventions for a specific environment.
+
+To address the largest use case, we are proposing a built-in Package Importer
+for the Node ecosystem. Our recommendation is for package authors to define a
+`sass` [conditional export] for entry points to their package in their
+distributed `package.json`. For example, a `package.json` containing:
+
+[conditional export]: https://nodejs.org/api/packages.html#conditional-exports
+
+```json
+{
+  "exports": {
+    ".": {
+      "sass": "./src/scss/index.scss",
+      "import": "./dist/js/index.mjs",
+      "default": "./dist/js/index.js"
+    },
+    "./utils": {
+      "sass": "./src/scss/_utils.scss",
+      "default": "./dist/js/utils.js"
+    }
+  }
+}
+```
+
+would allow a package consumer to write:
+
+```scss
+@use "pkg:library";
+@use "pkg:library/utils";
+```
+
+The Node Package Importer also supports several other ways of defining entry
+points, which it resolves in the following order:
```

Comment:
It's probably worth mentioning here that this covers the ways most Sass packages already define their entrypoints, so they can be loaded with `pkg:` without needing any changes. I know you mention that below, but it helps add context for why we support so many different mechanisms.

Reply:
Added in https://github.com/sass/sass-site/pull/848/commits/b5310ddf28c1425732fd93bdf4ae1447971ec54e